test/test_wordcloud.py

```diff
@@ -268,3 +271,19 @@ def test_unicode_stopwords():
 
     assert_true(words_unicode == words_str)
 
+def test_recolor_too_small():
+    # check exception is raised when image is too small
+    def _test():
+        colouring = np.array(Image.new('RGB', size=(20, 20)))
+        wc = WordCloud(width=30, height=30).generate(THIS)
+        image_colors = ImageColorGenerator(colouring)
+        wc.recolor(color_func=image_colors)
+    assert_raises_regex(ValueError, 'ImageColorGenerator is smaller than the canvas', _test)
```

Comment:
you could have just passed "wc.recolor" as the method, I think, but this way is fine, too.